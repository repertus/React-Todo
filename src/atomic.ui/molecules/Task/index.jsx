import React, { Component } from 'react';
import PropTypes from 'prop-types';

import Input from '../../atoms/Input/index';
import Label from '../../atoms/Label/index';
import { TaskDivStyling } from './index.styling';

class Task extends Component {
  render() {
    const { text } = this.props;

    return (
      <TaskDivStyling>
        <Label text={text} />
        <Input />
      </TaskDivStyling>
    );
  }
}

Task.propTypes = {
  text: PropTypes.string,
};

Task.defaultProps = {
  text: 'Task',
};

export default Task;
