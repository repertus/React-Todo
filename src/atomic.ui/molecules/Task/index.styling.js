import styled from 'styled-components';

export const TaskDivStyling = styled.div`
  margin-top: 20px;
`;
