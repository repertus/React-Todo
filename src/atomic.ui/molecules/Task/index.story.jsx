import React from 'react';
import { storiesOf } from '@storybook/react';

import Task from './index';

storiesOf('Molecules', module).add('Task Input', () => <Task />);
