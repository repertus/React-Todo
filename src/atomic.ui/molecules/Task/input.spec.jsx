import React from 'react';
import renderer from 'react-test-renderer';

import Task from './index';

describe('Task molecule tests', () => {
  test('Render simple select', () => {
    const component = renderer.create(<Task />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
