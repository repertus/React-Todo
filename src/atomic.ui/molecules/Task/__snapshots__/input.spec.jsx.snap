// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Task molecule tests Render simple select 1`] = `
<div
  className="sc-htpNat iuwWrn"
>
  <label
    className="sc-bwzfXH bCIaMc"
  >
    Task
  </label>
  <input
    className="sc-bdVaJa fWIDwd"
  />
</div>
`;
